tests/scripts/branch-history

```diff
@@ -0,0 +1,151 @@
+#!/bin/bash
+
+# Copyright (c) 2010, Lawrence Livermore National Security, LLC. Produced at the
+# Lawrence Livermore National Laboratory. LLNL-CODE-443211. All Rights reserved.
+# See file COPYRIGHT for details.
+#
+# This file is part of the MFEM library. For more information and source code
+# availability see http://mfem.org.
+#
+# MFEM is free software; you can redistribute it and/or modify it under the
+# terms of the GNU Lesser General Public License (as published by the Free
+# Software Foundation) version 2.1 dated February 1999.
+
+# Print usage information
+case $1 in
+   -h|-help)
+      cat <<EOF
+
+   $0 [-h|-help] {mfem_dir}
+
+   where: {mfem_dir}  is the MFEM source directory [default value: ../..]
+          -h|-help    prints this usage information and exits
+
+   This script checks if the current branch history, defined as the commits that
+   will be merged in master (those shown in a GitHub PR), contains unusually
+   large files, unusually large number of changes in a commit, unusually large
+   number of commits, etc.
+
+   The specific checks are provided at 3 levels: branch, commit and file.
+
+   Branch-level checks:
+   - Check if the branch contains unusually large number of commits
+
+   Commit-level checks:
+   - Check if the commit changes an unusually large number of files
+   - Check if the commit changes an unusually large number of lines
+
+   File-level checks:
+   - Check if the file is binary
+   - Check if the file is unusually large
+   - Check if an unusually large number of lines have been added to the file
+   - Check if an unusually large number of lines have been deleted from the file
+
```

Comment:
Updated documentation

Reply:
What the hell is going on here

Reply:
Slightly changed usage to separate the 3 cases.